txscript/script.go

```diff
@@ -851,10 +952,14 @@ func getWitnessSigOps(pkScript []byte, witness wire.TxWitness) int {
 // guaranteed to fail at execution.  This allows inputs to be pruned instantly
 // when entering the UTXO set.
 func IsUnspendable(pkScript []byte) bool {
-	pops, err := parseScript(pkScript)
+	// Not provably unspendable
+	if len(pkScript) == 0 {
+		return false
+	}
+	numberOfPops, firstOpcode, err := checkScriptTemplateParseable(pkScript, &opcodeArray)
 	if err != nil {
 		return true
 	}
 
-	return len(pops) > 0 && pops[0].opcode.value == OP_RETURN
+	return numberOfPops > 0 && firstOpcode == OP_RETURN
```

Comment:
How about using `(*byte, error)` as the return type of `checkScriptTemplateParseable`? The check would simply then be:

```go
firstOpcode, err := checkScriptTemplateParseable(pkScript, &opcodeArray)
if err != nil {
         return true
}

return firstOpcode != nil && *firstOpcode == OP_RETURN
```

This takes into account my other remark that if the script is empty, it doesn't make sense for the first opcode to be `OP_0`.